### ISSUE TITLE
Erratic magic animation sync

### ISSUE THREAD

**Issue by Humankiin** [tiltedphoques/TiltedEvolution#680]
## Version of Skyrim Together

**1.6.5**

## Description

Remote player's magic casting animations sync erratically. Sometimes their arms will raise, and sometimes they won't--in which case the magic will still fire, but their character will retain the magic out idle animation.

If casting with one hand, sometimes the other hand will also go down entirely to the player's side, as if they didn't have their hands out / were casting whatsoever.

## Steps to reproduce

1. Connect and join a party as normal.
2. Have remote player cast magic so you can watch them. (Specifically, flames is an easy one, as it is concentration.)
3. Make sure they try casting on and off, on and off, repeatedly. Holding magic in both hands makes it easy. Their hands will raise only some of the time.

-->

## Reproduction rate

- [x] Consistent: I can reproduce this bug by following the steps to reproduce 100% of the time (*every time*)
- [ ] Occasional: I can reproduce this bug by following the steps to reproduce around 50% of the time
- [ ] Rare: I can reproduce this bug by following the steps to reproduce around 10% or less of the time

## Evidence (optional)

https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExb2FqN3VwOHR3bWJ3eDI1ZDg4bGpxdXo3cTR5ZDV3djBtZmJscG15dyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/5s0Wc1SWFXYnAtjdFc/giphy-downsized-large.gif

Yes, this evidence is from the Alternate Start cell, because it made it easy to test w/o coc riverwood. SKSE is not enabled, and it is reproducible irregardless.
EDIT: just tried without ANY mods just to be sure, and the issue persists.
![image](https://github.com/tiltedphoques/TiltedEvolution/assets/45407313/b8be943a-a69a-40c3-b2a0-ffa16b376875)

**Comment by Rolothe5th** [comment]
thats due to lack of upper body sync which will never be implemented

**Comment by MostExcellent** [comment]
> thats due to lack of upper body sync which will never be implemented

not exactly, it is actually because magic behaviors are accessed from a separate node which often is not available when they are forced

**Comment by MostExcellent** [comment]
> > thats due to lack of upper body sync which will never be implemented
> 
> not exactly, I believe, from testing and some other investigation, that it is actually because magic behaviors are accessed from a separate behavior graph node (magicbehaviors.hkx) which often is not available when they are forced

Additionally, with mods, there can be...issues with respect to getting into a state in which *all* necessary nodes to actually start a casting animation are available

**Comment by MostExcellent** [comment]
Actually I think I have a fix, will review + push + pr when I have a chance
